Amplify/DevMenu/Data/PluginInfoHelper.swift

```diff
@@ -14,86 +14,70 @@ struct PluginInfoHelper {
+                makePluginInfoItem(for: $0.key, versionable: $0.value as? AmplifyVersionable)
             }
-        }
+        )
 
-        // Add Logging plugins information
-        if let versionable = Amplify.Logging.plugin as? AmplifyVersionable {
-            pluginList.append(PluginInfoItem(displayName: Amplify.Logging.plugin.key, information: versionable.version))
-        } else {
-            pluginList.append(PluginInfoItem(displayName: Amplify.Logging.plugin.key,
-                                             information: DevMenuStringConstants.versionNotAvailable))
-        }
+        pluginList.append(
+            makePluginInfoItem(
+                for: Amplify.Logging.plugin.key,
+                versionable: Amplify.Logging.plugin as? AmplifyVersionable
+            )
+        )
 
-        // Add Predictions plugins information
-        for pluginKey in Amplify.Predictions.plugins.keys {
-            if let versionable = (try? Amplify.Predictions.getPlugin(for: pluginKey)) as? AmplifyVersionable {
-                pluginList.append(PluginInfoItem(displayName: pluginKey, information: versionable.version))
-            } else {
-                pluginList.append(PluginInfoItem(displayName: pluginKey,
-                                                 information: DevMenuStringConstants.versionNotAvailable))
+        pluginList.append(contentsOf: Amplify.Predictions.plugins.map {
+                makePluginInfoItem(for: $0.key, versionable: $0.value as? AmplifyVersionable)
             }
-        }
+        )
 
-        // Add Storage plugins information
-        for pluginKey in Amplify.Storage.plugins.keys {
-            if let versionable = (try? Amplify.Predictions.getPlugin(for: pluginKey)) as? AmplifyVersionable {
-                pluginList.append(PluginInfoItem(displayName: pluginKey, information: versionable.version))
-            } else {
-                pluginList.append(PluginInfoItem(displayName: pluginKey,
-                                                 information: DevMenuStringConstants.versionNotAvailable))
+        pluginList.append(contentsOf: Amplify.Storage.plugins.map {
+                makePluginInfoItem(for: $0.key, versionable: $0.value as? AmplifyVersionable)
             }
-        }
+        )
 
         return pluginList
     }
+
+    private static func makePluginInfoItem(
+        for pluginKey: String,
+        versionable: AmplifyVersionable?
+    ) -> PluginInfoItem {
+        let pluginInfoItem: PluginInfoItem
+        if let versionable = versionable {
+            pluginInfoItem = PluginInfoItem(
+                displayName: pluginKey,
+                information: versionable.version
+            )
+        } else {
+            pluginInfoItem = PluginInfoItem(
+                displayName: pluginKey,
+                information: DevMenuStringConstants.versionNotAvailable
+            )
+        }
+        return pluginInfoItem
+    }
```

Comment:
guard statement?
```suggestion
   
      guard let versionable = versionable else {
           return PluginInfoItem(
                displayName: pluginKey,
                information: DevMenuStringConstants.versionNotAvailable
            )
        } 
        return PluginInfoItem(displayName: pluginKey, information: versionable.version)
    }
```

Reply:
Hmm. In this case, I'm less instinctively inclined to apply a `guard` because it's not actually an early exit. But thinking about it more, I like how it makes the "I need this info to do the happy path" intent easier to follow, so yeah! Guard! :)

Reply:
Thinking about it even *more* though, it's even cleaner to just worry about the version:

```swift
let version = versionable?.version ?? DevMenuStringConstants.versionNotAvailable
return PluginInfoItem(displayName: pluginKey, information: version)
```